xarray/backends/h5netcdf_.py

```diff
@@ -466,19 +493,23 @@ def open_datatree(
             driver=driver,
             driver_kwds=driver_kwds,
         )
+        # Check for a group and make it a parent if it exists
         if group:
             parent = NodePath("/") / NodePath(group)
         else:
             parent = NodePath("/")
 
         manager = store._manager
-        ds = open_dataset(store, **kwargs)
-        tree_root = DataTree.from_dict({str(parent): ds})
+
+        # Open root group with `xr.open_dataset()` and it to dictionary of groups
+        ds = open_dataset(filename_or_obj, **kwargs)
+        groups_dict = {str(parent): ds}
```

Comment:
Added this since we need to read the root group once and add it to the dictionary.

Reply:
Thanks! How would you suggest prepending the root path to `_inter_nc_groups`?

Reply:
You can use the `NodePath` object and pathlib-like syntax, e.g. `<first-half> / <second-half>`

https://github.com/pydata/xarray/blob/8c8d097816a70e35ef60de301503aa33f662857c/xarray/core/treenode.py#L26

Reply:
we can use [`prepend` from the `itertools` recipes](https://docs.python.org/3/library/itertools.html#itertools-recipes)

Reply:
> we can use [prepend from the itertools recipes](https://docs.python.org/3/library/itertools.html#itertools-recipes)

We could, that's a neat solution for a general iterable, but given that we know we're dealing with paths here wouldn't it be better to use the dedicated class we have for handling node paths? (This is not particularly important though)

Reply:
no, I meant that `_iter_nc_groups` yields nodes / node paths, and we can use `prepend` to prepend the root node's path (as a `NodePath`)

Reply:
I don't think I am completely following this thread. But what if instead of reopening the file_obj we open the store like: 
```suggestion
        ds = open_dataset(store, **kwargs)
        groups_dict = {str(parent): ds}
```

Reply:
that would work, except now we have a separate code path for the parent group compared to the child groups (and `open_dataset` applies the decoders if I'm not wrong?). Looking at this again, I now believe sticking `yield parent` in between these two lines could fix this in a much simpler way: https://github.com/pydata/xarray/blob/c508cc6a2e3000a9d87d2f8c611aae8733be07bf/xarray/backends/common.py#L134-L135